spinta/commands/write.py

```diff
@@ -621,6 +622,8 @@ async def validate_data(
                 if '_revision' in data.given:
                     raise exceptions.ManagedProperty(data.model, property='_revision')
             if data.prop:
+                if data.action == Action.UPSERT and data.saved.get(data.prop.name) is NA:
+                    check_type_value(data.prop.dtype, data.given.get(data.prop.name))
```

Comment:
This is already in `complex_data_check`, so I think these two lines should be removed. And I think, this should go into simple check.